src/game/server/ddracechat.cpp

```diff
@@ -1523,6 +1523,59 @@ void CGameContext::ConTele(IConsole::IResult *pResult, void *pUserData)
 	pChr->Core()->m_Vel = vec2(0, 0);
 }
 
+void CGameContext::ConPracticeUnSolo(IConsole::IResult *pResult, void *pUserData)
+{
+	CGameContext *pSelf = (CGameContext *)pUserData;
+	if(!CheckClientID(pResult->m_ClientID))
+		return;
+	CPlayer *pPlayer = pSelf->m_apPlayers[pResult->m_ClientID];
+	if(!pPlayer)
+		return;
+	CCharacter *pChr = pPlayer->GetCharacter();
+	if(!pChr)
+		return;
+
+	if(g_Config.m_SvTeam == SV_TEAM_FORBIDDEN || g_Config.m_SvTeam == SV_TEAM_FORCED_SOLO)
+	{
+		pSelf->SendChatTarget(pPlayer->GetCID(), "Command is not available on solo servers");
+		return;
+	}
+
+	CGameTeams &Teams = ((CGameControllerDDRace *)pSelf->m_pController)->m_Teams;
+	int Team = Teams.m_Core.Team(pResult->m_ClientID);
+	if(!Teams.IsPractice(Team))
+	{
+		pSelf->SendChatTarget(pPlayer->GetCID(), "You're not in a team with /practice turned on. Note that you can't earn a rank with practice enabled.");
+		return;
+	}
+
+	pChr->SetSolo(false);
+}
+
+void CGameContext::ConPracticeUnDeep(IConsole::IResult *pResult, void *pUserData)
```

Comment:
```suggestion
void CGameContext::ConPracticeUndeep(IConsole::IResult *pResult, void *pUserData)
```
I consider "undeep" and "unsolo" to be one verb, so no capitalization inbetween

Reply:
I agree it's better but I mainly copied the style from the other commands:

```c++
void CGameContext::ConUnSolo(IConsole::IResult *pResult, void *pUserData)
```
```c++
void CGameContext::ConUnDeep(IConsole::IResult *pResult, void *pUserData)
```

Should I rename the new commands to `Unsolo` and `Undeep` anyway?

Reply:
If you want, you can also rename the functions in the existing code (preferably in a separate commit or PR).

Reply:
@Robyt3 If you mean for the whole project, there are 60 results in 12 files for `Un[A-Z]`, which made me a little bit hesitant as I don't know how big of a change that refactoring would be (for custom modes and forks, developers used to the old syntax, branches in progress) considering it'd include changes to core functions like character `UnFreeze` for example. Also not sure whether terms like `UnFreezeHammer` would also be considered one verb and become `Unfreezehammer` (and whether that's a better style in this case).